test/modules/stats-widget/stats-widget.spec.js

```diff
@@ -112,11 +114,39 @@ test('StatsWidget#formatters', t => {
   statsWidget.setStats(stats);
 
   stats.get('Count').addCount(1000);
-  stats.get('Memory').addCount(1500);
+  stats.get('GPU Memory').addCount(1500);
   statsWidget.update();
 
   t.equals(statsWidget._items.Count.innerHTML, 'Count: 1.0k', 'Should use customized formatter.');
-  t.equals(statsWidget._items.Memory.innerHTML, 'Memory: 1500', 'Should use customized formatter.');
+  t.equals(
+    statsWidget._items['GPU Memory'].innerHTML,
+    'GPU Memory: 1500',
+    'Should use customized formatter.'
+  );
+
+  statsWidget.setStats(new Stats({id: 'test-stats-2'}));
+  t.equals(statsWidget._header.innerText, 'test-stats-2', "Should use the new stats' header.");
+
+  t.end();
+});
+
+test('StatsWidget#resetOnUpdate', t => {
+  const container = _global.document.createElement('div');
+  container.id = 'test-stats-widget-container';
+  const statsWidget = new StatsWidget(null, {
+    container,
+    resetOnUpdate: {Count: true}
```

Comment:
The `StatWidget.props.resetOnUpdate` interface seems non-optimal... wouldn't it would be more natural to set such flags directly on each stat definition in the stat object, rather than maintaining a separate map in the stats widget? 

```
new Stats({
  id, 
  stats: {
    count: {name: 'Count', type: 'count', resetOnUpdate: true}
  }
});
```

In addition, having this as an option on the stats widget again forces applications to know something about the stats inside the stats object, and adds extra lines of code clutter to apps that just want to present the stats in an "opaque" stats object in their UI.

We moved the types / formatting information to avoid apps having to specify it, now we are adding new things for apps to specify...